### PR TITLE
Improve Relative and formatted time calculation

### PR DESCRIPTION
**Pull request by tameremad**
* refactor a new component
* fix tests
* unify the usage of this Formatted Date component
* add tests
* more precision (minutes, hours, days... ago)

### CODE REVIEW

File: services/frontend-service/src/ui/components/LockDisplay/LockDisplay.test.tsx [hunk 1]
```diff
@@ -16,77 +16,11 @@ Copyright 2023 freiheit.com*/
 import { act, render } from '@testing-library/react';
 import { Spy } from 'spy4js';
 import { DisplayLock } from '../../utils/store';
-import { calcLockAge, daysToString, isOutdated, LockDisplay } from './LockDisplay';
+import { isOutdated, LockDisplay } from './LockDisplay';
 import { documentQuerySelectorSafe } from '../../../setupTests';
 const mock_addAction = Spy.mockModule('../../utils/store', 'addAction');
 
 describe('Test Auxiliary Functions for Lock Display', () => {
```
Comment (tameremad) [inline services/frontend-service/src/ui/components/LockDisplay/LockDisplay.test.tsx:23]: calcLockAge and daysToString were deleted. no need to test the same code twice

File: services/frontend-service/src/ui/components/ReleaseCard/ReleaseCard.test.tsx [hunk 2]
```diff
@@ -13,63 +13,12 @@ You should have received a copy of the MIT License
...
 
 Copyright 2023 freiheit.com*/
-import { getFormattedReleaseDate, ReleaseCard, ReleaseCardProps } from './ReleaseCard';
+import { ReleaseCard, ReleaseCardProps } from './ReleaseCard';
 import { render } from '@testing-library/react';
 import { UpdateOverview } from '../../utils/store';
 import { MemoryRouter } from 'react-router-dom';
 import { Environment, Release, UndeploySummary } from '../../../api/api';
 
-describe('Relative Date Calculation', () => {
```
Comment (tameremad) [inline services/frontend-service/src/ui/components/ReleaseCard/ReleaseCard.test.tsx:22]: this was moved to FormattedDate test

File: services/frontend-service/src/ui/components/FormattedDate/FormattedDate.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        createdAt.getDate()
+    )}`;
+    const relativeDate = useRelativeDate(createdAt);
+
+    // getHours automatically gets the hours in the correct timezone. in 24h format (no timezone calculation needed)
+    const formattedTime = `${createdAt.getHours()}:${createdAt.getMinutes()}`;
+
+    return (
+        <span className={className} title={createdAt.toString()}>
+            {formattedDate + ' @ ' + formattedTime + ' | '}
```
Comment (hannesg) [inline services/frontend-service/src/ui/components/FormattedDate/FormattedDate.tsx:76]: I don't think we need to display the full date and the relative date and have the full date again on hover. I would suggest to display only the relative time, without the seperate <i> tag and have the full time on hover.

**Review by hannesg — APPROVED** [review]

File: services/frontend-service/src/ui/components/FormattedDate/FormattedDate.tsx (hunk 3 above)
Comment (tameremad) [reply services/frontend-service/src/ui/components/FormattedDate/FormattedDate.tsx]: done :+1:

**Pull request merged**